src/Catel.MVVM/Catel.MVVM.Shared/MVVM/ViewModels/ViewModelBase.cs

```diff
@@ -1477,10 +1492,19 @@ protected virtual Task InitializeAsync()
             {
                 lock (_modelLock)
                 {
+                    bool isDirty = false;
                     foreach (KeyValuePair<string, object> modelKeyValuePair in _modelObjects)
                     {
                         UninitializeModelInternal(modelKeyValuePair.Key, modelKeyValuePair.Value, ModelCleanUpMode.EndEdit);
+
+                        var model = modelKeyValuePair.Value as IModel;
+                        if (model != null)
+                        {
+                            isDirty |= model.IsDirty;
+                        }
                     }
+
+                    IsDirty = isDirty;
```

Comment:
Not sure if we need to set IsDirty here, we are saving the vm here anyway. If that's not the case, then what's SavedAsync doing below?